packages/trie/src/db/flat/stackTrie.ts

```diff
@@ -0,0 +1,251 @@
+    this._value = value
+  }
+
+  insert(key: Nibbles, value: Uint8Array): BaseNode {
+    const commonLen = matchingNibbleLength(this._key, key)
+    assert.ok(commonLen < this._key.length, 'replacing leaf is not supported')
+
+    const newLeaf = new STLeafNode(key.slice(commonLen + 1), value)
+
+    //eslint-disable-next-line
+    const branch = new STBranchNode()
+    branch.setChild(key[commonLen], newLeaf)
+    branch.setChild(this._key[commonLen], this)
+
+    let root: BaseNode = branch
+    // Need extension node for common path
+    if (commonLen > 0) {
+      //eslint-disable-next-line
+      root = new STExtensionNode(this._key.slice(0, commonLen), branch)
+    }
+
+    // Slice common part from current leaf's key
+    this._key = this._key.slice(commonLen + 1)
+
+    return root
+  }
+
+  raw(): [Uint8Array, Uint8Array] {
+    const val: Nibbles = this._key.slice(0)
+    const encodedKey = addHexPrefix(val, true) // add terminator value to end of key
+    return [nibbleTypeToPackedBytes(encodedKey), this._value]
+  }
+}
+
+export class STExtensionNode extends BaseNode {
+  _key: Nibbles
+  _child: BaseNode
+
+  constructor(key: Nibbles, child: BaseNode) {
+    super()
+    this._key = key
+    this._child = child
+  }
+
+  insert(key: Nibbles, value: Uint8Array): BaseNode {
+    const commonLen = matchingNibbleLength(this._key, key)
+    // If matching prefix, recurse into child node
+    if (this._key.length === commonLen) {
+      this._child = this._child.insert(key.slice(commonLen), value)
+      return this
+    }
+
+    // Otherwise we'll need a new leaf, a branch and possibly an extension
+    const newLeaf = new STLeafNode(key.slice(commonLen + 1), value)
+
+    //eslint-disable-next-line
+    const branch = new STBranchNode()
+    branch.setChild(key[commonLen], newLeaf)
+
+    if (commonLen < this._key.length - 1) {
+      branch.setChild(this._key[commonLen], this)
+      this._key = this._key.slice(commonLen + 1)
+    } else {
+      branch.setChild(this._key[commonLen], this._child)
+    }
+
+    let root: BaseNode = branch
+    // Need extension for common prefix
+    if (commonLen > 0) {
+      root = new STExtensionNode(key.slice(0, commonLen), branch)
+    }
+
+    return root
+  }
+
+  raw(): any {
+    const childRaw = this._child.raw()
+    const childSerialized = RLP.encode(childRaw)
+    const value = childSerialized.length < 32 ? childRaw : keccak256(childSerialized)
+    const encodedKey = addHexPrefix(this._key.slice(0), false)
+    return [nibbleTypeToPackedBytes(encodedKey), value]
+  }
+}
+
+// TODO: Either remove HashNode or precompute hash
+//       to save on iterating already visited sub-tries.
+export class HashNode extends BaseNode {
+  _ref: BaseNode
+  _serialized: Uint8Array
+
+  constructor(ref: BaseNode) {
+    super()
+    this._ref = ref
+    this._serialized = ref.serialize()
+  }
+
+  get embedded(): boolean {
+    return this._serialized.length < 32
+  }
+
+  // eslint-disable-next-line
```

Comment:
Can be removed
```suggestion
```

Reply:
It's necessary at this point because if it is removed, the trie package CI run will fail in the linting step and the tests won't be run: https://github.com/ethereumjs/ethereumjs-monorepo/actions/runs/7341554101/job/19989439562?pr=2980